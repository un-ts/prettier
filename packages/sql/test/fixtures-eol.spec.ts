import fs from 'node:fs'
import path from 'node:path'
import { fileURLToPath } from 'node:url'

import { format } from 'prettier'

import SqlPlugin, { type SqlFormatOptions } from 'prettier-plugin-sql'

const PARSER_OPTIONS: Record<string, SqlFormatOptions> = {
  556: {
    language: 'mysql',
    endOfLine: 'crlf',
  },
  557: {
    language: 'mysql',
    endOfLine: 'cr',
  },
  558: {
    language: 'mysql',
    endOfLine: 'lf',
  },
  559: {
    language: 'mysql',
    endOfLine: 'auto',
  },
}

const _dirname =
  typeof __dirname === 'undefined'
    ? path.dirname(fileURLToPath(import.meta.url))
    : __dirname

describe('parser and printer', () => {
  it('printer should handle eol correctly', async () => {
    const fixtures = path.resolve(_dirname, 'fixtures-eol')
    for (const filepath of fs.readdirSync(fixtures)) {
      const input = fs.readFileSync(path.resolve(fixtures, filepath)).toString()

      const caseName = filepath.slice(0, filepath.lastIndexOf('.'))

      try {
        const output = await format(input, {
          filepath,
          parser: 'sql',
          plugins: [SqlPlugin],
          ...PARSER_OPTIONS[caseName],
        })

        expect(JSON.stringify(output)).toMatchSnapshot(filepath)
      } catch (error) {
        expect(error).toMatchSnapshot(filepath)
      }
    }
  })
})
