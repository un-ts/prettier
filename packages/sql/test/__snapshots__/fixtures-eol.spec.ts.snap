// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parser and printer > printer should handle eol correctly > 556.sql 1`] = `""CREATE TABLE \\"test\\" (\\"id\\" uuid NOT NULL)\\r\\nWITH\\r\\n  (oids = false);\\r\\n""`;

exports[`parser and printer > printer should handle eol correctly > 557.sql 1`] = `""CREATE TABLE \\"test\\" (\\"id\\" uuid NOT NULL)\\rWITH\\r  (oids = false);\\r""`;

exports[`parser and printer > printer should handle eol correctly > 558.sql 1`] = `""CREATE TABLE \\"test\\" (\\"id\\" uuid NOT NULL)\\nWITH\\n  (oids = false);\\n""`;

exports[`parser and printer > printer should handle eol correctly > 559.sql 1`] = `""CREATE TABLE \\"test\\" (\\"id\\" uuid NOT NULL)\\r\\nWITH\\r\\n  (oids = false);\\r\\n""`;
